### PR TITLE
make kubermatic chart context aware

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Makes the use of a general values.yaml more easy.
Also i removed the datacenters secret.yaml

### CODE REVIEW

File: config/kubermatic/values.yaml [hunk 1]
```diff
@@ -1,53 +1,32 @@
...
+  # external domain for the kubermatic installation. For example 'dev.kubermatic.io'
+  domain: ""
+  # base64 encoded kubeconfig which gives admin access to all seed clusters
+  kubeconfig: ""
+  # DEPRECATED! the external port for the customer cluster apiservers
+  externalApiserverPort: "8443"
+  controller:
+    replicas: 2
+    image:
+      repository: "kubermatic/api"
```
Comment (alvaroaleman) [inline config/kubermatic/values.yaml:21]: Why not make the `repository` and `tag` a single string and split it on `:` when rendering?

File: config/kubermatic/templates/kubermatic-cluster-controller-dep.yaml [hunk 2]
```diff
@@ -24,31 +24,30 @@ spec:
...
+          - -kubeconfig="/opt/.kube/kubeconfig"
+          - -master-resources="/opt/master-files"
+          - -datacenters="/opt/datacenter/datacenters.yaml"
+          - -versions="/opt/master-files/versions.yaml"
+          - -updates="/opt/master-files/updates.yaml"
+          - -apiserver-external-port={{ .Values.kubermatic.externalApiserverPort }}
           - -prometheus-address=0.0.0.0:8085
           - -prometheus-path=/metrics
-          image: 'kubermatic/api:{{.Values.KubermaticControllerImageTag}}'
-          imagePullPolicy: Always
```
Comment (alvaroaleman) [inline config/kubermatic/templates/kubermatic-cluster-controller-dep.yaml:37]: Shouldn't we keep that especially if we frequently use `latest` as tag?

File: config/kubermatic/templates/kubermatic-api-dep.yaml [hunk 3]
```diff
@@ -41,19 +40,16 @@ spec:
               protocol: TCP
           volumeMounts:
             - name: kubeconfig
-              mountPath: {{.Values.KubeConfigMountPath}}
-              readOnly: true
-            - name: datacenter-secrets
-              mountPath: {{.Values.DatacenterSecretsMountPath}}
+              mountPath: "/opt/.kube"
```
Comment (alvaroaleman) [inline config/kubermatic/templates/kubermatic-api-dep.yaml:43]: Why not use `/etc` for configs? And why not put all kubermatic-related configs into a common folder?

**Review by alvaroaleman — CHANGES_REQUESTED** [review]

File: config/kubermatic/values.yaml (hunk 1 above)
Comment (mrIncompetent) [reply config/kubermatic/values.yaml:21]: for what benifit?
Its common practice to have them separated: https://github.com/kubernetes/charts/tree/master/stable

File: config/kubermatic/templates/kubermatic-cluster-controller-dep.yaml (hunk 2 above)
Comment (mrIncompetent) [reply config/kubermatic/templates/kubermatic-cluster-controller-dep.yaml:37]: Good point, i'll add a config var for the ImagePullPolicy

File: config/kubermatic/templates/kubermatic-api-dep.yaml (hunk 3 above)
Comment (mrIncompetent) [reply config/kubermatic/templates/kubermatic-api-dep.yaml:43]: Sure, i don't mind

File: config/kubermatic/values.yaml (hunk 1 above)
Comment (alvaroaleman) [reply config/kubermatic/values.yaml:21]: Well, "common practice" depends a lot on your context, for both Docker and Kubernetes it is not

File: config/kubermatic/templates/kubermatic-api-dep.yaml (hunk 3 above)
Comment (mrIncompetent) [reply config/kubermatic/templates/kubermatic-api-dep.yaml:43]: Just realized that kubermatic has some path's hardcoded.
I'll postpone changing the Path's

**Comment by mrIncompetent** [comment]
@alvaroaleman PTAL

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**